tidy3d/components/monitor.py

```diff
@@ -667,6 +667,68 @@ def storage_size(self, num_cells: int, tmesh: ArrayLike[float, 1]) -> int:
         return BYTES_COMPLEX * len(self.ux) * len(self.uy) * len(self.freqs) * 4
 
 
+class DiffractionEfficiencyMonitor(PlanarMonitor, FreqMonitor):
+    """:class:`Monitor` that uses a 2D Fourier transform to compute the
+    diffraction efficiency for given (or all) diffraction orders.
+
+    Example
+    -------
+    >>> monitor = DiffractionEfficiencyMonitor(
+    ...     center=(1,2,3),
+    ...     size=(inf,inf,0),
+    ...     freqs=[250e12, 300e12],
+    ...     name='diffraction_efficiency_monitor',
+    ...     normal_dir='+',
+    ...     orders_x=[0, 1, 2],
+    ...     orders_y=[0],
+    ...     )
+    """
+
+    normal_dir: Direction = pydantic.Field(
+        "+",
+        title="Normal vector orientation",
+        description="Direction of the surface monitor's normal vector w.r.t. "
+        "the positive x, y or z unit vectors. Must be one of ``'+'`` or ``'-'``. "
+        "Defaults to ``'+'`` if not provided.",
+    )
+
+    orders_x: Tuple[int, ...] = pydantic.Field(
+        [0],
+        title="Diffraction orders along x",
+        description="Diffraction orders along x for which efficiency should be computed. "
+        "Orders corresponding to wave numbers outside of the light cone will be ignored. "
+        "By default, only order 0 will be considered.",
+    )
+
+    orders_y: Tuple[int, ...] = pydantic.Field(
+        [0],
+        title="Diffraction orders along y",
+        description="Diffraction orders along y for which efficiency should be computed. "
+        "Orders corresponding to wave numbers outside of the light cone will be ignored. "
+        "By default, only order 0 will be considered.",
+    )
+
+    @pydantic.validator("size", always=True)
+    def diffraction_efficiency_size(cls, val):
+        """Ensure that the monitor is infinite in the transverse direction."""
+        if val.count(inf) != 2:
+            raise SetupError(
+                "A 'DiffractionEfficiencyMonitor' must have a size of `td.inf` along both "
```

Comment:
let's use regular (single) quotes around `td.inf`. like you have for `DiffractionEfficiencyMonitor` in this same line.